src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -520,6 +520,10 @@ void HSHomeObject::on_shard_meta_blk_recover_completed(bool success) {
     std::unordered_set< homestore::chunk_num_t > excluding_chunks;
     std::scoped_lock lock_guard(_pg_lock);
     for (auto& pair : _pg_map) {
+        if (dynamic_cast< HS_PG* >(pair.second.get())->pg_state_.is_state_set(PGStateMask::DISK_DOWN)) {
```

Comment:
when we recover shard_metablk in line 516, should we check when the pg of the shard has a state of `PGStateMask::DISK_DOWN`? if yes, skip it.